dart/lib/src/transport/http_transport.dart

```diff
@@ -50,6 +50,10 @@ class HttpTransport implements Transport {
     if (response.statusCode == 200) {
       return _parseEventId(response);
     }
+    if (response.statusCode == 429) {
+      // throw error if rate limit is reached
+      throw QuotaExceededException();
```

Comment:
in the SDK generally we don't throw exceptions, but rather log it 

e.g `options.logger(
      SentryLevel.warning, 'example status code message')`
      
+ this might cause a recursion since we override `FlutterError.onError` and try to send unhandled errors where this Exception will be triggered again and so on etc..